packages/flutter/lib/src/material/bottom_sheet.dart

```diff
@@ -103,6 +103,8 @@ class _ModalBottomSheetLayout extends OneChildLayoutDelegate {
     childTop.end = childSize.height;
     return new Point(0.0, size.height - childTop.value);
   }
+
+  bool shouldRelayout() => true;
```

Comment:
Suppose the caller rebuilds and pases the same parameters to the layout delegate.  In that case, a layout isn't required.  Take a look at some of the implementation of CustomPainter to see how they implement shouldRelayout.

In the specific case of _ModalBottomSheet, imagine that after the animation is done, there's an app-wide setState that triggers every component in the app to rebuild.  In that case, we'll rebuild the CustomOneChildLayout but no new layout is needed.  The way you've written the code, you'll cause a layout.

The solution is create a new _ModalBottomSheetLayout object every time _ModalBottomSheetState goes through build and pass the concrete value of `childTop.value` to _ModalBottomSheetLayout.  Then shouldRelayout can compare the double it has for childTop with the double the oldDelegate has to determine whether a new layout is needed.  That will correctly handle the animation case and the case of an app-wide rebuild.